### ISSUE TITLE
[Discover-Next][UI] Query editor can be collapsed

### ISSUE THREAD

**Issue by kavilla** [opensearch-project/OpenSearch-Dashboards#7038]
### Description

When query enhancements are enabled. Users default see the editor expanded, but users select the `arrowDown` icon
![Screenshot 2024-06-15 at 5 43 55 PM](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/12060704/fcd1fa0a-6e82-44ef-b364-f69283901dca)

The query editor should collapse for query string input type of component.

### Requirements
* Get mocks for collapsed view
* Questions for UX:
   * Confirm we display an ellipsis (...) if query is too long
   * Can users still change the query language in collapsed view?
   * Can users modify the query in the collapsed view? Or when they select/focus the input it automatically expand to the query editor?
   * Dashboards and visualizations will have the default bar. The `arrowDown` icon is tied to the saved query. If we are making expanding the default. Should we consider removing the [icon globally for saved query](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/plugins/data/public/ui/saved_query_management/saved_query_management_component.tsx#L185) (regardless of this is toggled on or not)
      * ![Screenshot 2024-06-15 at 6 09 06 PM](https://github.com/opensearch-project/OpenSearch-Dashboards/assets/12060704/d00e2580-b154-4306-b284-1927ed645953)
* Implement
   * Design Consideration: Depending on the mocks, what design proposal could be expanding between the query editor and query string input component. But if there are a lot of differences between query string input and collapsed view then implement a different component.
   * tl;dr: might need a design proposal
* Tests
